Basic items docs

12awsomeman34 opened the pull request:
Nobody seems to have done this yet, so here it is.

List to exit WIP:
- [x] Update item data docs to the value api
- [x] Item Properties
- [x] NBT

12awsomeman34 committed:
Basic items docs

Whoops

Updates on wording

Updates on wording (again)

Basic items docs

Whoops

Updates on wording

Updates on wording (again)

RobertHerhold commented on source/plugin/basics/items.rst:
```diff
@@ -0,0 +1,61 @@
+        return type == ItemTypes.STICK;
+    }
+    
+See how simple that is? Now we know that our item is a stick, but sticks can stack, so how many of them are there?
```
"Now we know that our item is a stick, but sticks can stack, so how many of them are there?"

Run-on sentence - maybe "Because sticks can stack, we can also find out how many are present"

RobertHerhold commented:
Please also differentiate between ItemStacks and Item entities

RobertHerhold commented on source/plugin/basics/items.rst:
```diff
@@ -0,0 +1,61 @@
+
+    import org.spongepowered.api.item.inventory.ItemStack;
+    
+    public int getAmount(ItemStack stack) {
```
Useless method here?

RobertHerhold commented on source/plugin/basics/items.rst:
```diff
@@ -0,0 +1,61 @@
+        return stack.getQuantity();
+    }
+    
+Modifying Item Data of an ItemStack
```
This is a bit verbose - maybe "Modifying ItemStack Data"

gabizou commented on source/plugin/basics/items.rst:
```diff
@@ -0,0 +1,61 @@
+    import com.google.common.base.Optional;
+
+    public void setUnbreakable(ItemStack stack) {
+        Optional<DurabilityData> optional = stack.getData(DurabilityData.class);
```
If possible, I'd hold off on writing the bit about Data since it's going to be changed for the better by the end of the week (I'm completing implementation for usability this week).

gabizou replied:
So, IRL stuff happened, haven't had any productive work in the last 4 days. I'm going to try and get the Values API merge worthy by Friday for the sake of making the lives of everyone easier.

RobertHerhold commented on source/plugin/basics/items.rst:
```diff
@@ -0,0 +1,50 @@
+Working with Items
+==================
+
+Items are represented through an ``ItemStack``. An ``ItemStack`` is an inventory item with information such as the amount of the item in the stack, the type of the item, and extra data such as durability. An item itself is the graphical representation of an ``ItemStack`` as an entity.
```
"An item itself..." - please make "Item" capitalized and in an in-line code block: `Item`, as it is a class that you are referring to.

Tzky committed:
refactor migrating pages

* merged migrating pages
* reduced nestes pages
* added overview to new migrating page

Tzky committed:
streamlining the faq

* rework the faq
* re-add faq overview on top of page

Tzky committed:
[WIP] Condense contributing section

This PR cleans up the mess in /contributing/ leaving just a few pages
in.

This is achieved by removing old (empty) files, merging SpongeForge and
Vanilla into a single implementation page and replacing the Docs files
with a single page.

Comments on this are appreciated. I'll probably create another PR if
there are other sections that need revision too.

RobertHerhold committed:
Merge pull request #316 from SpongePowered/streamline/contributing

Condense contributing section

RobertHerhold committed:
Merge pull request #315 from SpongePowered/streamline/faq-rework

Clean up FAQ

Tzky committed:
Merge pull request #314 from SpongePowered/streamline/create-server

Refactor migrating pages

RobertHerhold committed:
getTaskBuilder() --> createTaskBuilder()

Inscrutable committed:
added added

Inscrutable committed:
SpongeAPI not Sponge API

RobertHerhold committed:
Unify documentation (no basic/advanced categories)

gabizou committed:
Update Spawning Entity code example

This updates the spawning entity code example with the recent Data API
changes.

Signed-off-by: Gabriel Harris-Rouquette <gabizou@me.com>

Inscrutable committed:
Retired Tyr

RobertHerhold committed:
Merge branch 'unification'

12awsomeman34 committed:
Git is fum

12awsomeman34 committed:
Git is not fum

gabizou commented on source/plugin/items.rst:
```diff
@@ -0,0 +1,50 @@
+    
+    public boolean isStick(ItemStack stack) {
+        ItemType type = stack.getItem();
+        return type == ItemTypes.STICK;
```
should check `type.equals(ItemTypes.STICK)` instead.

RobertHerhold commented:
@12AwsomeMan34 
Do you plan on working more on this?

12AwsomeMan34 commented:
Yeah, got caught up with irl issues. I'll see what I can get done by this weekend.

12awsomeman34 committed:
type.equals

12awsomeman34 committed:
Item Data 2.0

gabizou commented on source/plugin/items.rst:
```diff
@@ -0,0 +1,57 @@
+    import org.spongepowered.api.item.inventory.ItemStack;
+
+    public void setUnbreakable(ItemStack stack) {
+        stack.offer(Keys.ITEM_DURABILITY, -1);
```
This is totally not how it works.... There's a Key for UNBREAKABLE.

12AwsomeMan34 replied:
I checked before and didn't see it, so I assumed that -1 durability was unbreakable. Oh well, quick fix.

12awsomeman34 committed:
Quick fix

I done messed up (again)

12awsomeman34 committed:
Quick fix 2

12awsomeman34 committed:
Item Properties

12awsomeman34 committed:
Java 8 optional

12awsomeman34 committed:
Small note of data translators

Inscrutable commented:
Is there anything else to add @12AwsomeMan34 or is this about ready to roll?

12AwsomeMan34 commented:
I'd say this is as good as it's going to get.

saladoc commented:
@12AwsomeMan34 Looks good. I'm just doing some polishing, then I'll merge.
